src/app/data-planes/data/index.ts

```diff
@@ -141,9 +165,9 @@ const DataplaneInsight = {
 export const DataplaneOverview = {
   fromObject(partialDataplaneOverview: PartialDataplaneOverview, canUseZones: boolean): DataplaneOverview {
     const dataplaneInsight = DataplaneInsight.fromObject(partialDataplaneOverview.dataplaneInsight)
-    const networking = DataplaneNetworking.fromObject(partialDataplaneOverview.dataplane.networking)
 
-    const dataplaneType = getDataplaneType(networking)
+    const networking = DataplaneNetworking.fromObject(partialDataplaneOverview.dataplane.networking, typeof dataplaneInsight.connectedSubscription !== 'undefined')
```

Comment:
**Question**: Could you explain this addition a bit, please?

How come the health property becomes dependent on data from `dataplaneInsight` here?

Reply:
My question is about the business details of this change and not about code organization. So when I say "the health property becomes dependent on data from `dataplaneInsight`" that is definitely the case: for the single inbound here, ready will change depending on whether there is a connected subscription as you explain here:

> In "filling it out" we don't want to always hardcode the health `ready: true` and therefore show it as "online" because in the case of gateways thats not true if there is no `connectedSubscription`. Therefore we pass in a `true`/`false` based on whether there is a `connectedSubscription`.

My question is basically: does that make sense?

This doesn’t happen anywhere else. We don’t change the health object of inbounds if a regular Dataplane has no connected subscription so I find it weird to have this influence here. If we show an individual inbound’s readyness in places, it’s confusing to me that sometimes, the presence of a connected subscription would change the readyness status and sometimes it wouldn’t.

Reply:
Just to be clear because you mention a "regular Dataplane" are you asking:

> does that make sense for sidecars and gateways?

or

> does that make sense for gateways?

?

Reply:
The latter:

> does that make sense for gateways?

Reply:
Ok thanks. What single thing do we use currently to understand the status of a gateway dataplane?

Reply:
The presence of a connected subscription. But what we don’t do, is make that information influence an unrelated property (the health object on inbounds).

Reply:
Ok:

1. We don't use this `builtin.inbound.health.ready` information anywhere as yet, its currently just a base default.
2. In order to show "builtin gateway status" everywhere in the GUI we use the presence of a connected subscription.
3. We have a choice between two options, `builtin.inbound.health.ready:` `true` or `false`. We don't want a three-way boolean. We either pick one based on nothing, or we pick one based on something _as a default_. To be honest in this case if we didn't want to use whether the dataplane is connected or not _as a default_ for inbound health, I'd probably rather it was `false`.

A slightly related question, if the dataplane isn't connected to the dataplane, how do you know the health of the inbound?

Reply:
> A slightly related question, if the dataplane isn't connected, how do you know the health of the inbound?

I don’t know.

I guess let’s resolve this, but I definitely think it’s weird that we write to an object here that we usually use exclusively as the source of data. The fact that we don’t have this data from the get go kind of implies there’s no meaningful information here that could be called "inbound health". It’s extra weird that this behaves differently depending on whether we "make up" this single inbound or use inbounds from the API resource.